python/ribasim/ribasim/input_base.py

```diff
@@ -241,8 +241,20 @@ def _write_table(self, temp_path: Path) -> None:
         """
         table = self.tablename()
         assert self.df is not None
+
+        # Add `fid` to all tables as primary key
+        # Enables editing values manually in QGIS
+        df = self.df.copy()
```

Comment:
Why can't we add the column in place?

Reply:
Because you don't want it to be part of the model. If a user wants to do other operations on the dataframe, you don't want to bother them with extra columns.